src/hotspot/cpu/arm/arm_32.ad

```diff
@@ -432,8 +432,7 @@ OptoRegPair c2::return_value(int ideal_reg) {
 
 int MachCallStaticJavaNode::ret_addr_offset() {
   bool far = (_method == nullptr) ? maybe_far_call(this) : !cache_reachable();
-  return ((far ? 3 : 1) + (_method_handle_invoke ? 1 : 0)) *
-    NativeInstruction::instruction_size;
+  return (far ? 3 : 1) * NativeInstruction::instruction_size;
 }
```

Comment:
Why do we still need the `instruction_size` offset? Are all static java calls now method handles?

Reply:
The offset is in bytes, so we still need to convert from instruction count to bytes with instruction_size.  This change adjusts for the fact that method handle calls have 1 fewer instruction on arm32 now, because preserve_SP was removed.